Sources/StreamChatUI/Utils/ImageLoader.swift

```diff
@@ -0,0 +1,115 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import Nuke
+import UIKit
+
+/// ImageLoading is providing set of functions for downloading of images from URLs.
+public protocol ImageLoading: AnyObject {
+    /// Load an image from the given URL
+    /// - Parameters:
+    ///   - url: The URL of the image
+    ///   - imageCDN: The CDN to be used for downloading the image
+    ///   - resize: Whether the image should be resized
+    ///   - preferredSize: The preferred size of the image to be downloaded
+    ///   - completion: Completion that gets called when the download is finished
+    @discardableResult
+    func loadImage(
+        from url: URL,
+        imageCDN: ImageCDN,
+        resize: Bool,
```

Comment:
Let's add `ImageCDN` as a dependency of `DefaultImageLoader` and make the `Components.imageLoader` lazy var